tests/Functional/Admin/SharedBlockAdminTest.php

```diff
@@ -45,11 +45,151 @@ public function testCrudUrls(string $url, array $parameters = []): void
+            'type' => 'sonata.block.service.text',
+        ], 'btn_create_and_list', [
+            'shared_block[name]' => 'Name',
+            'shared_block[enabled]' => 1,
+            'shared_block[settings][content]' => 'Text',
+        ]];
+
+        yield 'Create Shared Block - Template' => ['/admin/tests/app/sonatapageblock/shared/create', [
+            'uniqid' => 'shared_block',
+            'type' => 'sonata.block.service.template',
+        ], 'btn_create_and_list', [
+            'shared_block[name]' => 'Name',
+            'shared_block[enabled]' => 1,
+            'shared_block[settings][template]' => '@SonataBlock/Block/block_template.html.twig',
+        ]];
+
+        yield 'Create Shared Block - Menu' => ['/admin/tests/app/sonatapageblock/shared/create', [
+            'uniqid' => 'shared_block',
+            'type' => 'sonata.block.service.menu',
+        ], 'btn_create_and_list', [
+            'shared_block[name]' => 'Name',
+            'shared_block[enabled]' => 1,
+            'shared_block[settings][title]' => 'Title',
+            'shared_block[settings][menu_name]' => 'sonata_admin_sidebar',
+            'shared_block[settings][safe_labels]' => 1,
+            'shared_block[settings][current_class]' => 'active',
+            'shared_block[settings][first_class]' => 'first',
+            'shared_block[settings][last_class]' => 'last',
+            'shared_block[settings][menu_class]' => 'menu_sonata',
+            'shared_block[settings][children_class]' => 'children',
+            'shared_block[settings][menu_template]' => '@SonataBlock/Block/block_core_menu.html.twig',
+        ]];
+
+        yield 'Create Shared Block - RSS' => ['/admin/tests/app/sonatapageblock/shared/create', [
+            'uniqid' => 'shared_block',
+            'type' => 'sonata.block.service.rss',
+        ], 'btn_create_and_list', [
+            'shared_block[name]' => 'Name',
+            'shared_block[enabled]' => 1,
+            'shared_block[settings][url]' => 'https://sonata-project.org',
+            'shared_block[settings][title]' => 'Title',
+            'shared_block[settings][translation_domain]' => 'SonataPageBundle',
+            'shared_block[settings][icon]' => 'fa fa-home',
+            'shared_block[settings][class]' => 'custom_class',
+        ]];
+
+        // Blocks From SonataPageBundle
+        yield 'Create Shared Block - Children Pages' => ['/admin/tests/app/sonatapageblock/shared/create', [
+            'uniqid' => 'shared_block',
+            'type' => 'sonata.page.block.children_pages',
+        ], 'btn_create_and_list', [
+            'shared_block[name]' => 'Name',
+            'shared_block[enabled]' => 1,
+            'shared_block[settings][title]' => 'Title',
+            'shared_block[settings][translation_domain]' => 'SonataPageBundle',
+            'shared_block[settings][icon]' => 'fa fa-home',
+            'shared_block[settings][current]' => 1,
+            // 'shared_block[settings][pageId]' => 1,
```

Comment:
Should we remove this line or uncomment it ?

Reply:
There is a pageId on the settings, but not sure how to make it populate with pages, because it needs a page on the block. I leave it commented to not forget to think about it...